app/src/main/java/io/github/droidkaigi/confsched2018/presentation/NavigationController.kt

```diff
@@ -137,9 +138,14 @@ class NavigationController @Inject constructor(private val activity: AppCompatAc
                     intent.putExtra(Intent.EXTRA_REFERRER, appUri)
                 }
 
-        // welcome contributions :)
-        // e.g. support in-app browser
+        val packageName = CustomTabsHelper.getPackageNameToUse(activity)
+        packageName ?: run {
+            // Cannot use custom tabs.
+            activity.startActivity(customTabsIntent.intent.setData(Uri.parse(url)))
```

Comment:
Should we check browser availability eagerly?

Reply:
Hmm, you mean like this? I don't think it's necessary 🤔 

```kotlin
try {
  val packageName = CustomTabsHelper.getPackageNameToUse(activity)
  packageName ?: run {
    activity.startActivity(customTabsIntent.intent.setData(Uri.parse(url)))
    return
  }
  customTabsIntent.intent.`package` = packageName
  customTabsIntent.launchUrl(activity, Uri.parse(url))
} catch (e: ActivityNotFoundException) {
  // Show original webview activity
}
```